src/test/scala/com/wavesplatform/state/diffs/SetScriptTransactionDiffTest.scala

```diff
@@ -70,4 +71,36 @@ class SetScriptTransactionDiffTest extends PropSpec with PropertyChecks with Mat
         }
     }
   }
+
+  property("Script with BlockV2 only works after Ride4DApps feature activation") {
+    import com.wavesplatform.lagonaki.mocks.TestBlock.{create => block}
+
+    val settingsUnactivated = TestFunctionalitySettings.Enabled.copy(
+      preActivatedFeatures = Map(
+        BlockchainFeatures.Ride4DApps.id -> 3
+      ))
+    val settingsActivated = TestFunctionalitySettings.Enabled.copy(
+      preActivatedFeatures = Map(
+        BlockchainFeatures.Ride4DApps.id -> 0
+      ))
+    val setup = for {
+      master <- accountGen
+      ts     <- positiveLongGen
+      genesis = GenesisTransaction.create(master, ENOUGH_AMT, ts).explicitGet()
+      expr    = BLOCKV2(LET("x", CONST_LONG(3)), CONST_BOOLEAN(true))
+      script  = ScriptV1(V1, expr, checkSize = false).explicitGet()
+      tx      = SetScriptTransaction.selfSigned(1, master, Some(script), 100000, ts + 1).explicitGet()
+    } yield (genesis, tx)
+
+    forAll(setup) {
+      case (genesis, tx) =>
+        assertDiffEi(Seq(block(Seq(genesis))), block(Seq(tx)), settingsUnactivated) { blockDiffEi =>
+          blockDiffEi should produce("SetScriptTransaction has not been activated yet")
```

Comment:
The error should be different. It's not SetScriptTransaction that has not been activated(it has), but rather RIDE4DApps feature that enables BlockV2 has not been activated yet. Mind the signature of  `activationBarrier`, it allows for that.